core/src/session.rs

```diff
@@ -251,16 +267,12 @@ impl Session {
 }
 
 #[derive(Clone)]
-pub struct SessionWeak(Weak<SessionInternal>);
+struct SessionWeak(Weak<SessionInternal>);
```

Comment:
Can this be removed all together?

Reply:
No, the dispatcher task needs a weak reference too. I was looking for another solution without one, but I didn't find one that convinced me.